client/admin/sidebar/useSettingsGroupsFiltered.js

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useMemo, useState } from 'react';
+import { useDebouncedValue } from '@rocket.chat/fuselage-hooks';
+
+import { PrivateSettingsCachedCollection } from '../PrivateSettingsCachedCollection';
+import { useTranslation } from '../../contexts/TranslationContext';
+import { settings } from '../../../app/settings';
+
+
+export const useSettingsGroupsFiltered = (textFilter) => {
+	const t = useTranslation();
+	const [collection, setCollection] = useState(settings.collectionPrivate);
+
+	const filter = useDebouncedValue(textFilter, 400);
+
+	useEffect(() => {
+		(async () => {
```

Comment:
More an opinion than a correction or suggestion... I think IIFEs using anonymous functions are not well-suited in so nested scopes like this one because they are harder to debug in this context. An IIFE with named function (`(async function loadData() { ... })()`) or a arrow function declaration followed by its call (`const loadData = async () => { ... }; loadData();`) will preserve the function name, making easier to debug stack traces.